tests/microos/live_patching.pm

```diff
@@ -0,0 +1,100 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: smoke test for Live Patching in SLE Micro
+#           1) Get current kernel version
+#           2) Make sure the system has been registered previously
+#           3) Register Live Patching module if not done already
+#           4) Optionally add a kernel update repo
+#           5) Update system and reboot
+#           6) Get new kernel version and compare it
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use transactional;
+use utils qw(zypper_call);
+use version_utils qw(is_sle_micro);
+
+sub register_lp_module {
+    my $arch        = get_required_var('ARCH');
+    my $regcode     = get_required_var('SCC_REGCODE');
+    my $sle_version = get_required_var('VERSION_SLE');
+    my $extensions  = script_output('SUSEConnect --list-extensions');
+    record_info('extensions', $extensions);
+    unless ($extensions =~ m/Live Patching.*Activated/) {
+        record_info('INFO', 'Registering module "sle-module-live-patching"');
+        trup_call("register -p sle-module-live-patching/$sle_version/$arch -r $regcode");
+        check_reboot_changes;
+        $extensions = script_output('SUSEConnect --list-extensions');
+        record_info('extensions', $extensions);
+        die('There was a problem activating the Live Patching module') unless ($extensions =~ m/Live Patching.*Activated/);
+    }
+}
+
+sub configure_lp {
+    assert_script_run("sed -i 's/multiversion =.*/multiversion = provides:multiversion(kernel)/' /etc/zypp/zypp.conf");
+    assert_script_run("sed -i \'s/multiversion.kernels =.*/multiversion.kernels = latest/g\' /etc/zypp/zypp.conf");
+    assert_script_run("echo \"LIVEPATCH_KERNEL='always'\" > /etc/sysconfig/livepatching");
+}
+
+sub setup_optional_repo {
+    # Option to use a test repository with an existing kernel update.
+    # This is useful if regular channels don't provide any kernel update
+    # and we want to force testing this feature
+    my $test_repo = get_var('SLE_MICRO_KERNEL_REPO');
+    if ($test_repo) {
+        record_info('INFO', "Adding repository $test_repo");
+        assert_script_run 'curl -k https://ca.suse.de/certificates/ca/SUSE_Trust_Root.crt -o /etc/pki/trust/anchors/SUSE_Trust_Root.crt';
```

Comment:
Could use `ensure_ca_certificates_suse_installed` instead. Needs to be adjusted to work with read-only / as well by copying this there